client/src/components/CheckoutForm.js

```diff
@@ -38,6 +41,18 @@ const CheckoutForm = () => {
             </div>
           </Grid>
         </Grid>
+
+        <Grid item>
+          <Button
+            color="primary"
+            variant="contained"
+            className={classes.button}
+          >
```

Comment:
this will redirect you to home because you're logged in.
Will cause a redirection "/"->"/signup"->"/home".

I think it should be  <Link to="/checkout" className={classes.link}>